src/main/java/de/tum/in/www1/artemis/repository/metrics/ExerciseMetricsRepository.java

```diff
@@ -0,0 +1,124 @@
+package de.tum.in.www1.artemis.repository.metrics;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.util.Set;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ExerciseInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ScoreDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.SubmissionTimestampDTO;
+
+/**
+ * Spring Data JPA repository to fetch exercise related metrics.
+ */
+@Profile(PROFILE_CORE)
+@Repository
+public interface ExerciseMetricsRepository extends JpaRepository<Exercise, Long> {
+
+    /**
+     * Get the exercise information for all exercises in a course.
+     *
+     * @param courseId the id of the course
+     * @return the exercise information for all exercises in the course
+     */
+    @Query("""
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ExerciseInformationDTO(e.id, e.shortName, e.title, COALESCE(e.startDate, e.releaseDate), e.dueDate, e.class)
+            FROM Exercise e
+            WHERE e.course.id = :courseId
+            """)
+    Set<ExerciseInformationDTO> findAllExerciseInformationByCourseId(long courseId);
+
+    @Query("""
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ScoreDTO(e.id, COALESCE(AVG(r.score), 0))
+            FROM Exercise e
+                LEFT JOIN StudentParticipation p ON e.id = p.exercise.id
+                LEFT JOIN p.submissions s
+                LEFT JOIN s.results r
+            WHERE e.id IN :exerciseIds
+                AND s.submitted = TRUE
+            GROUP BY e.id
+            """)
+    Set<ScoreDTO> findAverageScore(@Param("exerciseIds") Set<Long> exerciseIds);
```

Comment:
Do we really want the average of all scores and not only the lastest score of each participation?